_core/smart/principals.md

```diff
@@ -14,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
 
 ## Principals and tx-sender
 
-A principal is represented by a public-key hash or multi-signature Stacks address. Assets in Clarity and the Stacks blockchain are "owned" by objects of the principal type; put another way, principal object types may own an asset. 
-
-A given principal operates on its assets by issuing a signed transaction on the Stacks blockchain. A Clarity contract can use a globally defined `tx-sender` variable to obtain the current principal.
-
-The following user-defined function transfers an asset, in this case, tokens, between two principals:
-
-```
-(define (transfer! (sender principal) (recipient principal) (amount int))
-  (if (and
-        (not (eq? sender recipient))
-        (debit-balance! sender amount)
-        (credit-balance! recipient amount))
-    'true
-    'false))
+Assets in the smart contracting language and blockchain are
+"owned" by objects of the principal type, meaning that any object of
+the principal type may own an asset. For the case of public-key hash
+and multi-signature Stacks addresses, a given principal can operate on
+their assets by issuing a signed transaction on the blockchain. _Smart
+contracts_ may also be principals (reprepresented by the smart
+contract's identifier), however, there is no private key associated
+with the smart contract, and it cannot broadcast a signed transaction
+on the blockchain.
+
+A Clarity contract can use a globally defined `tx-sender` variable to
+obtain the current principal. The following example defines a transaction
+type that transfers `amount` uSTX from the sender to a recipient if amount
+is a multiple of 10, otherwise returning a 400 error code.
+
+```scheme
+(define-public (transfer-to-recipient! (recipient principal) (amount uint))
+  (if (is-eq (mod amount 10) 0)
+      (stx-transfer? amount tx-sender recipient)
+      (err u400)))
 ```
 
-The principal's signature is not checked by the smart contract, but by the virtual machine.
-
-
 ## Smart contracts as principals
 
-Smart contracts themselves are principals and are represented by the smart contract's identifier. You create the identifier when you launch the contract, for example, the contract identifier here is `hanomine`.
+Smart contracts themselves are principals and are represented by the
+smart contract's identifier -- which is the publishing address of the
+contract _and_ the contract's name, e.g.:
 
-```bash
-clarity-cli launch hanomine /data/hano.clar /data/db
+```scheme
+'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR.contract-name
 ```
 
-A smart contract may use the special variable `contract-name` to refer to its own principal.
+For convenience, smart contracts may write a contract's identifier in the
+form `.contract-name`, which is expanded by the Clarity interpreter into
+a fully-qualified contract identifier that corresponds to the same
```

Comment:
This is a mouthful - can we explain in simpler terms and also add why this is relevant? what's the outcome of this behavior?

Reply:
I expanded on this a bit in 76b9d152

Reply:
thanks!